geode-core/src/test/java/org/apache/geode/distributed/ServerStateTest.java

```diff
@@ -0,0 +1,207 @@
+    assertThat(value.getPort()).isEqualTo(port);
+    assertThat(value.getServiceLocation()).isEqualTo(serviceLocation);
+    assertThat(value.getStatus().getDescription()).isEqualTo(statusDescription);
+    assertThat(value.getStatusMessage()).isEqualTo(statusMessage);
+    assertThat(value.getTimestamp().getTime()).isEqualTo(timestampTime);
+    assertThat(value.getUptime()).isEqualTo(uptime);
+    assertThat(value.getWorkingDirectory()).isEqualTo(workingDirectory);
+  }
+
+
+  @Test
+  public void fromJsonWithValidJsonStringReturnsServerStateToStringNoStatusMessage() {
+    // given: valid json string
+    String jsonString = createStatusJson();
+
+    // when: passed to fromJson
+    ServerLauncher.ServerState value = fromJson(jsonString);
+
+    String toString = value.toString();
+    assertThat(toString.contains(TO_STRING_STATUS_MESSAGE)).isFalse();
+
+  }
+
+  @Test
+  public void fromJsonWithValidJsonStringReturnsServerStateToStringHasStatusMessage() {
+    // given: valid json string
+    statusDescription = AbstractLauncher.Status.STARTING.getDescription();
+    String jsonString = createStatusJson();
+
+    // when: passed to fromJson
+    ServerLauncher.ServerState value = fromJson(jsonString);
+
+    String toString = value.toString();
+    assertThat(toString.contains(TO_STRING_STATUS_MESSAGE)).isTrue();
+
+  }
+
+  private ServerLauncher.ServerState fromJson(final String value) {
+    return ServerLauncher.ServerState.fromJson(value);
+  }
+
+  private List<String> getJvmArguments() {
+    List<String> list = new ArrayList<>();
+    list.add(jvmArguments);
+    return list;
+  }
+
+  private String createStatusJson() {
+    Map<String, Object> map = new HashMap<>();
+    map.put(AbstractLauncher.ServiceState.JSON_CLASSPATH, classpath);
+    map.put(AbstractLauncher.ServiceState.JSON_GEMFIREVERSION, gemFireVersion);
+    map.put(AbstractLauncher.ServiceState.JSON_HOST, host);
+    map.put(AbstractLauncher.ServiceState.JSON_JAVAVERSION, javaVersion);
+    map.put(AbstractLauncher.ServiceState.JSON_JVMARGUMENTS, getJvmArguments());
+    map.put(AbstractLauncher.ServiceState.JSON_LOCATION, serviceLocation);
+    map.put(AbstractLauncher.ServiceState.JSON_LOGFILE, logFile);
+    map.put(AbstractLauncher.ServiceState.JSON_MEMBERNAME, memberName);
+    map.put(AbstractLauncher.ServiceState.JSON_PID, pid);
+    map.put(AbstractLauncher.ServiceState.JSON_PORT, port);
+    map.put(AbstractLauncher.ServiceState.JSON_STATUS, statusDescription);
+    map.put(AbstractLauncher.ServiceState.JSON_STATUSMESSAGE, statusMessage);
+    map.put(AbstractLauncher.ServiceState.JSON_TIMESTAMP, timestampTime);
+    map.put(AbstractLauncher.ServiceState.JSON_UPTIME, uptime);
+    map.put(AbstractLauncher.ServiceState.JSON_WORKINGDIRECTORY, workingDirectory);
+
+    String status = null;
+    try {
+      status = new ObjectMapper().writeValueAsString(map);
+    } catch (JsonProcessingException e) {
+      e.printStackTrace();
+    }
```

Comment:
Please add `throws JsonProcessingException` to the test method and remove the try-catch block.

Reply:
Thanks for comments. Updated.